src/python_testing/TC_CLCTRL_5_1.py

```diff
@@ -0,0 +1,734 @@
+        logging.info(f"FeatureMap: {feature_map}")
+        is_is_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kInstantaneous
+        is_ps_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kPositioning
+        is_cl_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kCalibration
+        is_lt_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kMotionLatching
+        is_pt_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kProtection
+        is_mo_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kManuallyOperable
+
+        # STEP 2b: If the IS feature is supported on the cluster, skip remaining steps and end test case.
+        self.step("2b")
+
+        if is_is_feature_supported:
+            logging.info("Instantanous feature is supported, skipping test case")
+
+            # Skipping all remainig steps
+            for step in self.get_test_steps(self.current_test_info.name)[self.current_step_index:]:
+                self.step(step.test_plan_number)
+                logging.info("Test step skipped")
+
+            return
+
+        # STEP 2c: TH reads TestEventTrigger attribute from the General Diagnostic Cluster
+        self.step("2c")
+
+        logging.info("Checking if TestEventTrigger attribute is enabled")
+        self.check_test_event_triggers_enabled()
+
+        # STEP 2d: TH reads from the DUT the (0xFFFB) AttributeList attribute
+        self.step("2d")
+
+        attribute_list = await self.read_clctrl_attribute_expect_success(endpoint, attributes.AttributeList)
+        logging.info(f"AttributeList: {attribute_list}")
+
+        # STEP 2e: TH establishes a wildcard subscription to all attributes on the Closure Control Cluster, with MinIntervalFloor = 0, MaxIntervalCeiling = 30 and KeepSubscriptions = false
+        self.step("2e")
+
+        sub_handler = ClusterAttributeChangeAccumulator(Clusters.ClosureControl)
+        await sub_handler.start(dev_controller, self.dut_node_id, endpoint=endpoint, min_interval_sec=0, max_interval_sec=30, keepSubscriptions=False)
+
+        # STEP 3a:If the LT feature is not supported on the cluster, skip steps 3a to 3j.
+        self.step("3a")
+
+        if is_lt_feature_supported:
+            logging.info("Motion Latching feature supported.")
+
+            # STEP 3b: If the attribute is supported on the cluster, TH reads from the DUT the OverallCurrentState attribute
+            self.step("3b")
+
+            if attributes.OverallCurrentState.attribute_id in attribute_list:
+                overall_current_state = await self.read_clctrl_attribute_expect_success(endpoint, attributes.OverallCurrentState)
+                logging.info(f"OverallCurrentState: {overall_current_state}")
+                if overall_current_state is None:
+                    logging.error("OverallCurrentState is None")
```

Comment:
If we do not assert and exit here, then use of `overall_current_state` in next line will give an error.
```suggestion
                    logging.error("OverallCurrentState is None")
                    asserts.assert_true(False, "OverallCurrentState is None.")
                    return
```